### ISSUE TITLE
False positives for at-variables prefixed with negative sign in value-keyword-case

### ISSUE THREAD

**Issue by edg2s** [stylelint/stylelint#2404]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

bug

> Which rule, if any, is this issue related to?

value-keyword-case

> What CSS is needed to reproduce this issue?

```less
.foo {
  width: @myWidth;
  height: -@myHeight;
  margin: 0 - @myMargin;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "value-keyword-case": "lower"
  }
}
```

> Which version of stylelint are you using?

7.8.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Node API (grunt-stylelint)

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

LESS

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

`Expected "-@myHeight" to be "-@myheight"`

The positive value `@myWidth` doesn't trigger an error, nor does the expression `0 - @myMargin`.

**Comment by edg2s** [comment]
See also #2406

**Comment by jeddy3** [comment]
@edg2s FYI, addressing non-standard syntax issues is not a priority for the core team. As such, please consider contributing a fix if you'd like to see this issue resolved.

**Comment by edg2s** [comment]
It looks like the issue here is the valueParser, it doesn't break up expressions unless there are spaces, so while `'0 - @foo'` parses as `['0', ' ', '-', ' ', '@foo']` and each one gets detected as a non-keyword, `'0-@foo'` parses as one node: `['0-@foo']` which can't be easily checked as a keyword

**Comment by evilebottnawi** [comment]
@edg2s can you create issue in `value-parser` repo?

**Comment by edg2s** [comment]
I think the issue is that `value-parser` is specifically designed to only work on regular CSS syntax (as it is currently defined). It seems there is another parser [postcss-values-parser](https://github.com/lesshint/postcss-values-parser/) designed just for this very purpose so I am experimenting with that.

**Comment by evilebottnawi** [comment]
Done in https://github.com/stylelint/stylelint/pull/2558
